### ISSUE TITLE
readers.copc forces at least 4 threads even if fewer are configured

### ISSUE THREAD

**Issue by wonder-sk** [PDAL/PDAL#3913]
COPC reader currently forces >= 4 threads in the thread pool, even if fewer have been requested:
https://github.com/PDAL/PDAL/blob/master/io/CopcReader.cpp#L190

Forcing just one thread by editing source code & recompiling worked fine, so either this limitation should be relaxed, or the documentation should be updated to reflect the limit.

Testing with a 10M COPC file on my laptop (SSD drive), reading performance does not improve beyond 3 threads, and with the default of 15 threads reading is even ~10% slower. It may be useful to note in the documentation that for datasets on a local file system the default of 15 threads may be too high.

**System/installation information:**

PDAL master (to be 2.5)

**Comment by abellgithub** [comment]
This came from EPT and I don't know that it makes sense (the min of 4). The assumption was that you were getting data from a remote connection. I did some testing and found that the best throughput dependsedon matching threads to your connection speed. I did some work trying to automate the number of threads based on responses from the remote server, but ultimately, the complexity really didn't buy much. I may have spent some time playing with SSD devices and thread count, but it all got really complicated so I think we just said we'd leave it to the user. Perhaps @connormanning has something to say about the 4 thread minimum since he stuck that in for EPT years ago.
